src/topology/maps.lean

```diff
@@ -456,40 +460,43 @@ lemma open_embedding_of_embedding_open {f : α → β} (h₁ : embedding f)
   (h₂ : is_open_map f) : open_embedding f :=
 ⟨h₁, h₂.is_open_range⟩
 
+lemma open_embedding_iff_embedding_open {f : α → β} :
+  open_embedding f ↔ embedding f ∧ is_open_map f :=
+⟨λ h, ⟨h.1, h.is_open_map⟩, λ h, open_embedding_of_embedding_open h.1 h.2⟩
+
 lemma open_embedding_of_continuous_injective_open {f : α → β} (h₁ : continuous f)
   (h₂ : function.injective f) (h₃ : is_open_map f) : open_embedding f :=
 begin
-  refine open_embedding_of_embedding_open ⟨⟨_⟩, h₂⟩ h₃,
-  apply le_antisymm (continuous_iff_le_induced.mp h₁) _,
-  intro s,
-  change is_open _ → is_open _,
-  rw is_open_induced_iff,
-  refine λ hs, ⟨f '' s, h₃ s hs, _⟩,
-  rw preimage_image_eq _ h₂
+  simp only [open_embedding_iff_embedding_open, embedding_iff, inducing_iff_nhds, *, and_true],
+  exact λ a, le_antisymm (h₁.tendsto _).le_comap
+    (@comap_map _ _ (𝓝 a) _ h₂ ▸ comap_mono (h₃.nhds_le _))
 end
 
+lemma open_embedding_iff_continuous_injective_open {f : α → β} :
+  open_embedding f ↔ continuous f ∧ function.injective f ∧ is_open_map f :=
+⟨λ h, ⟨h.continuous, h.inj, h.is_open_map⟩,
+  λ h, open_embedding_of_continuous_injective_open h.1 h.2.1 h.2.2⟩
+
 lemma open_embedding_id : open_embedding (@id α) :=
 ⟨embedding_id, is_open_map.id.is_open_range⟩
 
 lemma open_embedding.comp {g : β → γ} {f : α → β}
   (hg : open_embedding g) (hf : open_embedding f) : open_embedding (g ∘ f) :=
 ⟨hg.1.comp hf.1, (hg.is_open_map.comp hf.is_open_map).is_open_range⟩
 
-lemma open_embedding_of_open_embedding_compose {α β γ : Type*} [topological_space α]
-  [topological_space β] [topological_space γ] (f : α → β) {g : β → γ} (hg : open_embedding g)
-    (h : open_embedding (g ∘ f)) : open_embedding f :=
-begin
-  have hf := hg.to_embedding.continuous_iff.mpr h.continuous,
-  split,
-  { exact embedding_of_embedding_compose hf hg.continuous h.to_embedding },
-  { rw [hg.open_iff_image_open, ← set.image_univ, ← set.image_comp, ← h.open_iff_image_open],
-    exact is_open_univ }
-end
+lemma open_embedding.is_open_map_iff {g : β → γ} {f : α → β} (hg : open_embedding g) :
+  is_open_map f ↔ is_open_map (g ∘ f) :=
+by simp only [is_open_map_iff_nhds_le, ← @map_map _ _ _ _ f g, ← hg.map_nhds_eq,
+  map_le_map_iff hg.inj]
+
+lemma open_embedding_iff_open_embedding_compose (f : α → β) {g : β → γ} (hg : open_embedding g) :
```

Comment:
```suggestion
lemma open_embedding_comp_iff (f : α → β) {g : β → γ} (hg : open_embedding g) :
```

Reply:
Looking again, this should maybe be named to match [function.injective.of_comp_iff](https://leanprover-community.github.io/mathlib_docs/logic/function/basic.html#function.injective.of_comp_iff)